internal/dependencyinjection/container.go

```diff
@@ -0,0 +1,73 @@
+package dependencyinjection
+
+import (
+	"context"
+	"sync"
+
+	"github.com/stellar/stellar-disbursement-platform-backend/db"
+)
+
+var (
+	// dependenciesStore is the global store for all the service instances.
+	dependenciesStore = make(map[string]interface{})
```

Comment:
Any reason why we wouldn't use  `var dependenciesStore sync.Map` ? It's also used in `MultiTenantDataSourceRouter`. That would avoid having to separately manage a mutex.

Reply:
It's because of the new deletion methods. I wanted to make `Get->Close->Delete` atomic for database connectionPools, but I couldn't do that with sync.Map because I can't manipulate its mutex directly.